src/deploy-cloudrun.ts

```diff
@@ -37,12 +37,61 @@ export async function run(): Promise<void> {
     // Flags
     const envVars = core.getInput('env_vars'); // String of env vars KEY=VALUE,...
     const region = core.getInput('region') || 'us-central1';
+    const source = core.getInput('source'); // Source directory
+    const suffix = core.getInput('suffix');
+    const tag = core.getInput('tag');
+    const noTraffic =
+      core.getInput('no_traffic').toLowerCase() == 'true' ? true : false;
+    const revTraffic = core.getInput('revision_traffic');
+    const tagTraffic = core.getInput('tag_traffic');
+    const flags = core.getInput('flags');
 
     let installBeta = false; // Flag for installing gcloud beta components
     let cmd;
+    // Throw errors if inputs aren't valid
+    if (revTraffic && tagTraffic) {
+      throw new Error(
+        'Both `revision_traffic` and `tag_traffic` inputs set - Please select one.',
+      );
+    }
+    if ((revTraffic || tagTraffic) && !name) {
+      throw new Error('No service name set.');
+    }
 
     // Find base command
-    if (metadata) {
+    if (revTraffic || tagTraffic) {
+      // Update traffic
+      cmd = [
+        'beta',
+        'run',
+        'services',
+        'update-traffic',
+        name,
+        '--platform',
+        'managed',
+        '--region',
+        region,
+      ];
+      installBeta = true;
+      if (revTraffic) cmd.push('--to-revisions', revTraffic);
+      if (tagTraffic) cmd.push('--to-tags', tagTraffic);
+    } else if (source) {
+      // Deploy service from source
+      cmd = [
+        'beta',
+        'run',
+        'deploy',
+        name,
+        '--quiet',
+        '--platform',
+        'managed',
+        '--region',
+        region,
+        '--source',
+        source,
+      ];
+      installBeta = true;
```

Comment:
Should we extract setting the beta in cmd into a common func where we can keep a list of beta commands? This way we can keep track of it in a single place when gcloud promotes to non beta.

Reply:
I like this idea, but I am not sure how that would be implemented.

Reply:
We could keep a list of beta flags in a list and do a lookup before we exec to check if any cmd args is in beta list.